src/abstractdataframe/show.jl

```diff
@@ -257,7 +257,8 @@ function getchunkbounds(maxwidths::Vector{Int},
         for j in 1:ncols
             # Include 2 spaces + | character in per-column character count
             totalwidth += maxwidths[j] + 3
-            if totalwidth > availablewidth
+            # Ensure there'd also be enough space to print the " ⋯" ellipses if needed
+            if totalwidth + (j < ncols ? 2 : 0) > availablewidth
```

Comment:
Can you please test the case when the width is exactly as the number of columns in the terminal. This is a problematic corner case on some terminals (not all; in particular the default Widnows terminal has this problem, but e.g. the new Windows terminal does not have this problem):
![image](https://user-images.githubusercontent.com/6187170/88427500-53820b80-cdf3-11ea-9df9-23bb3a02244d.png)
The strategy that should be used is to make it 1 char narrower, so that even the terminals that do not correctly handle the newline when they reach full width display the result correctly.

Maybe the fix you propose covers it but I would prefer to be sure :).

Reply:
Yes, the existing tests do a good job of covering this case and I added a very minimal extension.  Printing this data frame:

https://github.com/JuliaData/DataFrames.jl/pull/2330/commits/3fcdf6ed35a8ce3f99a998803aa3e439ae767750#diff-21ef36be0213824dd286b390e3590742R62

uses exactly 40 chars for the first 3 columns without the ellipses.  Thus with a displaysize of 40, it'll only print 2 columns if the output is elided, and it'll print 3 if all columns are printed (split).  I also added a test with a display size of 42 to ensure it prints 3 columns with the ellipses.

Reply:
This patch, though, doesn't give us that one extra buffer character.  I can easily add it here now if you'd like.

Reply:
> doesn't give us that one extra buffer character

This is what I would prefer please (otherwise we will be getting the problem with display as on the screenshot above on some terminals). This was a long standing issue, but we have not worked on improving display in DataFrames.jl for a while, but I think we should fix it if we touch this part of code.

Reply:
Done in https://github.com/JuliaData/DataFrames.jl/pull/2330/commits/ed3464ef45c6729e918d4050c615b64168ec2bd9

We'll see if my blind attempt at adjusting the tests worked...